pkg/reader/ccip.go

```diff
@@ -367,11 +367,23 @@ func (r *ccipChainReader) MsgsBetweenSeqNums(
 			return nil, fmt.Errorf("failed to cast %v to Message", item.Data)
 		}
 
-		msg.Message.ExtraArgsDecoded, err = r.extraDataCodec.DecodeExtraData(msg.Message.ExtraArgs, sourceChainSelector)
+		msg.Message.ExtraArgsDecoded, err = r.extraDataCodec.DecodeExtraArgs(msg.Message.ExtraArgs, sourceChainSelector)
 		if err != nil {
 			return nil, fmt.Errorf("failed to decode ExtraArgs: %w", err)
 		}
+
 		msg.Message.Header.OnRamp = onRampAddress
+
+		for i, tokenAmount := range msg.Message.TokenAmounts {
+			tokenAmount.DestExecDataDecoded, err = r.extraDataCodec.DecodeTokenAmountDestExecData(
```

Comment:
nit: maybe you can directly do this:
msg.Message.TokenAmounts[i].DestExecDataDecoded, err = ...